### ISSUE TITLE
istio 1.4.0 - istioctl manifest apply not working as expected

### ISSUE THREAD

**Issue by greenenergy** [istio/istio#19080]
**Bug description**
When executing istioctl manifest apply commands, order matters in an unexpected way, but is not mentioned in the documentation. Specifically, I configured my minikube installation according to the instio setup instructions, to configure a demo environment. I'm also using an nginx ingress to forward calls into the istio ingressgateway.

When I applied the manifests in the order mentioned in the documentation, it worked. I shortened the invocation from 3 separate calls to one which combined the commandline parameters (which I thought was fine to do, since they don't appear to conflict) it ran without error, but created a gateway that always failed with a 503 "bad gateway" error.

**Expected behavior**
That after applying the istio manifests described in the documentation, that a functioning ingressgateway would be provided.

**Steps to reproduce the bug**
The setup documentation at  [here](https://istio.io/docs/setup/install/istioctl/#install-istio-using-the-default-profile) lists example `manifest apply` commands. If you execute them in the sequence presented in the documentation, you end up with a functioning ingressgateway, like so:
```
  yes | istioctl manifest apply && \
    istioctl manifest apply --set values.global.mtls.enabled=true --set values.global.controlPlaneSecurityEnabled=true && \
    istioctl manifest apply --set profile=demo
```

But if you shorten this to one invocation wtih all those parameters, the ingressgateway will fail to forward any requests, and return a `503 bad gateway` error instead:

```
yes | istioctl manifest apply --set profile=demo --set values.global.mtls.enabled=true --set values.global.controlPlaneSecurityEnabled=true
```
My assumption when reading the docs & looking at the commands is that when you invoke `istioctl manifest apply` you are updating the current configuration to enable or disable something based on commandline parameters. However, the behaviour seems to indicate something else is going on - that every time you apply, it's going to add some things and remove some things.

If this is expected, then it would be nice to update the docs to reflect this fact, and perhaps explain it.

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**

client version: 1.4.0
control plane version: 1.4.0
data plane version: 1.4.0 (4 proxies)

$ kubectl version
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.4", GitCommit:"67d2fcf276fcd9cf743ad4be9a9ef5828adc082f", GitTreeState:"clean", BuildDate:"2019-09-18T14:51:13Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14+", GitVersion:"v1.14.7-gke.23", GitCommit:"81c87c699557fed991e292cd328b2129c2f242a2", GitTreeState:"clean", BuildDate:"2019-11-07T19:23:23Z", GoVersion:"go1.12.11b4", Compiler:"gc", Platform:"linux/amd64"}

**How was Istio installed?**
$ curl -L https://istio.io/downloadIstio | sh -
Then the `manifest apply` commands.

**Environment where bug was observed (cloud vendor, OS, etc)**
Minikube v1.5.2, Linux mint 18.1

**Comment by sdake** [comment]
I believe this is parallel to the addons not starting up properly.

Here is my brief analysis of what is happening:
1. CRDs are registeredd
1. operator waits for CRDs to finish registering
1. every other component is rendered and fed into the kubeapi interface.

That last one isn't quite right.  The ordering of application should **ALWAYS** be:
1. Any CRDs
1. Wait on CRDs
1. Any Custom Resources
1. Wait on Custom Resources
1. RBAC rules
1. Wait on RBAC rules
1. configuration objects (configmaps, secrets)
1. wait on configuration objects

There are probably some startup ordering that could use some tuning in the above, but that should get us moving in the right direction.

The real key is waiting on custom resources.  Currently when a manifest is rendered, custom resources may be used after they are needed, leading to a race (in mixer for example) - that blocks mixer (and as a result telemetry).

Cheers
-steve

**Comment by esnible** [comment]
My understanding is that when the user does

```
istioctl manifest apply --set values.global.mtls.enabled=true --set values.global.controlPlaneSecurityEnabled=true
istioctl manifest apply --set profile=demo
```

... the work to set mTLS and control plane security is overwritten.

When the user does 

```
istioctl manifest apply --set profile=demo --set values.global.mtls.enabled=true --set values.global.controlPlaneSecurityEnabled=true
```

They get mTLS and control plane security.  (For this user it isn't working for some reason.)

The user believed that setting `--set values.global.mtls.enabled=true` would be remembered between invocations.  That doesn't happen; there is a contentious issue to remember the current config on the cluster ( https://github.com/istio/istio/issues/20000 ).

**Comment by esnible** [comment]
I believe `manifest apply` AKA `upgrade` will work as expected if we get a fix for https://github.com/istio/istio/issues/23183 so that the default can be the current Istio settings for an  Istio installation instead of the default built-in IOP.

**Comment by esnible** [comment]
I will close this as a duplicate of https://github.com/istio/istio/issues/23183
